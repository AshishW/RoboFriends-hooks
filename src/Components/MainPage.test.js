import MainPage from "./MainPage";
import React from "react";
import {shallow, ShallowWrapper} from 'enzyme';

let wrapper;
beforeEach(()=>{
  const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps}/>);
}) //before each test, this function runs

it('renders the MainPage without crashing', ()=>{
    expect(wrapper).toMatchSnapshot()
})

it('filters robots correctly', ()=>{
  const mockProps2 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 1,
      name: 'test',
      email: 'test@gmail.com'
    }],
    searchField: 'test',
    isPending: false
  }
  
  const wrapper2 = shallow(<MainPage {...mockProps2}/>);
  expect(wrapper2.instance().filteredRobots()).toEqual([{
    id: 1,
    name: 'test',
    email: 'test@gmail.com'
  }])
})

it('filters robots correctly 2', ()=>{
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [{
      id: 1,
      name: 'test',
      email: 'test@gmail.com'
    }],
    searchField: 'a',
    isPending: false
  }
  
  const wrapper3 = shallow(<MainPage {...mockProps3}/>);
  expect(wrapper3.instance().filteredRobots()).toEqual([])
})

it('checks if the loading text appears when the robots are being requested', ()=>{
    const mockProps = {
      onRequestRobots: jest.fn(),
      robots: [],
      searchField: '',
      isPending: true
  }
  wrapper = shallow(<MainPage {...mockProps} />);
  expect(wrapper.text()).toEqual("LOADING...")
})