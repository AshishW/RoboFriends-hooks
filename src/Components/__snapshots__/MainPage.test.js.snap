// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MainPage without crashing 1`] = `ShallowWrapper {}`;
